src/main/java/reiff/projectile/ProjectileFrame.java

```diff
@@ -0,0 +1,160 @@
+        JLabel nextToXILabel = new JLabel();
+        JLabel emptySpace = new JLabel();
+        JButton calculateButton = new JButton("Calculate");
+
+
+        JSlider angleSlider = new JSlider(JSlider.HORIZONTAL, 0, 90, 0);
+        angleSlider.setMajorTickSpacing(10);
+        angleSlider.setMinorTickSpacing(1);
+        angleSlider.setPaintTicks(true);
+        angleSlider.setPaintLabels(true);
+
+
+        JLabel angleValueLabel = new JLabel("0");
+
+        angleSlider.addChangeListener(new ChangeListener() {
+            public void stateChanged(ChangeEvent e) {
+                JSlider source = (JSlider) e.getSource();
+                if (!source.getValueIsAdjusting()) {
+                    int angleValue = source.getValue();
+                    angleValueLabel.setText(String.valueOf(angleValue));
+                    recalculateValues();
+                }
+            }
+        });
+
+        JTextField velocityField = new JTextField();
+        velocityField.getDocument().addDocumentListener(new DocumentListener() {
+
+            @Override
+            public void insertUpdate(DocumentEvent e) {
+                recalculateValues();
+            }
+
+            @Override
+            public void removeUpdate(DocumentEvent e) {
+                recalculateValues();
+            }
+
+            @Override
+            public void changedUpdate(DocumentEvent e) {
+                recalculateValues();
+            }
+        });
+
+
+        JTextField secondsField = new JTextField();
+        secondsField.getDocument().addDocumentListener(new DocumentListener() {
+            @Override
+            public void insertUpdate(DocumentEvent e) {
+                recalculateValues();
+            }
+
+            @Override
+            public void removeUpdate(DocumentEvent e) {
+                recalculateValues();
+            }
+
+            @Override
+            public void changedUpdate(DocumentEvent e) {
+                recalculateValues();
+            }
+        });
+
+        setLayout(new GridLayout(0, 2));
+
+        add(angleLabel);
+        add(angleSlider);
+        add(new JLabel());
+        add(angleValueLabel);
+
+        add(velocityLabel);
+        add(velocityField);
+
+        add(secondsLabel);
+        add(secondsField);
+
+        add(xLabel);
+        add(nextToXLabel);
+
+        add(yLabel);
+        add(nextToYLabel);
+
+        add(peakYLabel);
+        add(nextToPeakYLabel);
+
+        add(XILabel);
+        add(nextToXILabel);
+
+        add(emptySpace);
+        add(calculateButton);
+
+
+        calculateButton.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                recalculateValues();
+            }
+        });
+
+        recalculateValues();
+    }
+
+    private void recalculateValues() {
+        Projectile projectile = new Projectile(
+                angleSlider.getValue(),
```

Comment:
The error message says that `"this.angleSlider" is null` on this line.

Where do you create the JSlider?

Reply:
But you're not accessing the one created on line 42. 
You're accessing the variable (which is null) created on line 13.